File: java/src/com/esielkar/adapter/Adapter.java
package com.esielkar.adapter;

public class Adapter implements ExistingInterface {
    // Java 8 or higher - Lambdas
    private final Service adaptee;

    public Adapter(Service adaptee) {
        this.adaptee = adaptee;
    }

    @Override
    public final int method(int data) {
        return adaptee.method(String.valueOf(data));

    }

}

File: dart/design_patterns/bin/adapter/adapter.dart
import 'service.dart';

mixin ExistingInterface {
  double method(int data);
}

class Adapter implements ExistingInterface {
  final Service _adaptee;

  Adapter(Service adaptee) : _adaptee = adaptee;

  @override
  double method(int data) {
    return _adaptee.method(data.toString());
  }
}
